src/lib/zkstring.ts

```diff
@@ -0,0 +1,148 @@
+import { Bool, Circuit, Field, Poseidon } from '../snarky';
+import { CircuitValue, prop } from './circuit_value';
+
+export class Character extends CircuitValue {
+  @prop value: Field;
+
+  constructor(value: Field) {
+    super();
+
+    if (Character.validateInput(value)) {
+      this.value = value;
+    } else {
+      throw Error('Character initialization did not pass validation');
+    }
+  }
+
+  equals(char: Character): Bool {
+    return this.value.equals(char.value);
+  }
+
+  toField(): Field {
+    return this.value;
+  }
+
+  toString(): string {
+    const charCode = Number(this.value.toString());
+    return String.fromCharCode(charCode);
+  }
+
+  static fromString(str: string) {
+    const char = Field(str.charCodeAt(0));
+    return new Character(char);
+  }
+
+  /* 
+  TODO: Add support for more character sets
+  ASCII reference: https://en.wikipedia.org/wiki/ASCII#Printable_characters
+  */
+  static validateInput(value: Field): Boolean {
+    if (Circuit.inProver() || Circuit.inCheckedComputation()) {
+      const ret = Bool.and(
+        value.gte(Field(32)),
+        value.lte(Field(126))
+      );
+
+      return ret.toBoolean();
+    }
+
+    return true;
+  }
+};
+
+export class ZKString extends CircuitValue {
+  @prop values: Character[];
+
+  constructor(values: Character[]) {
+    super();
+
+    this.values = values;
+  }
+
+  append(str: ZKString): ZKString {
+    return new ZKString([...this.values, ...str.values])
```

Comment:
This works re constraints (it's just reorganizing existing variables). But it should take fixed-length strings and return a string with a different fixed length